java/ql/src/semmle/code/xml/WebXML.qll

```diff
@@ -130,3 +130,59 @@ class WebListenerClass extends WebXMLElement {
    */
   Class getClass() { result.getQualifiedName() = getValue() }
 }
+
+/**
+ * A `<security-constraint>` element in a `web.xml` file.
+ */
+class SecurityConstraint extends WebXMLElement {
+  SecurityConstraint() { getName() = "security-constraint" }
+
+  XMLElement getAuthConstraint() { result = getAChild("auth-constraint") }
+}
+
+/**
+ * A `<web-resource-collection>` element in a `web.xml` file,
+ * nested under a `<security-constraint>` element.
+ */
+class WebResourceCollectionClass extends WebXMLElement {
+  WebResourceCollectionClass() {
+    getName() = "web-resource-collection" and
+    getParent() instanceof SecurityConstraint
+  }
+
+  Class getClass() { result.getQualifiedName() = getValue() }
+
+  /**Gets the security constraint to which this belongs */
+  SecurityConstraint getDeclaringConstraint() { result = getParent() }
+
+  /** Returns all HTTP methods listed in this collection */
+  string getHTTPMethods() {
```

Comment:
This should probably be `getAHTTPMethod()` (or <code>getA<i>n</i>HTTPMethod()</code>?) (see also the [style guide](https://github.com/github/codeql/blob/master/docs/ql-style-guide.md#naming)) and similarly the documentation comment would need to be adjusted

Reply:
I beg to differ here. Yes, what you say is very much true. The `getAHTTPMethod()` and `getAnHTTPMethod` nomenclature applies when the call returns multiple results. However, in this case, I am not returning the actual `<http-method>` tag rather the value in that tag. Hence, to differentiaite, I named it `getHttpMethods`. 

I could however rename it to `getAHttpMethodValue`. Its a bit verbose but would satisfy the style guide. WDYT?

Reply:
`getAHttpMethodValue` sounds good, though I don't know what the maintainers of this project prefer (I am not a maintainer).

Reply:
I have made the change. Let's see what the team has to say.